code/datums/components/directional_attack.dm

```diff
@@ -0,0 +1,155 @@
+/datum/component/directional_attack
+	var/active = TRUE
+	var/datum/action/directional_attack_toggle/toggle_action
+
+/datum/component/directional_attack/Initialize()
+	. = ..()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	toggle_action = new()
+	var/toggle_path
+	toggle_path = .proc/living_activation_toggle
+	toggle_action.give_action(parent)
+	toggle_action.update_button_icon(active)
+	RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, toggle_path)
+	if(active)
+		RegisterSignal(parent, COMSIG_MOB_CLICKON, .proc/select_directional_action)
+
+/datum/component/directional_attack/Destroy(force, silent)
+    QDEL_NULL(toggle_action)
+    return ..()
+
+/datum/component/directional_attack/proc/select_directional_action(datum/source, atom/A, params)
+	if(ishuman(parent))
+		human_directional_action(A, params)
+	else if(isxeno(parent))
+		xeno_directional_action(A, params)
```

Comment:
To cut down one check. please make the `isxeno()` check be the first `if`.

Considering you are only adding it to xenos at this point in time.